src/Export/VTK.jl

```diff
@@ -113,46 +235,150 @@ function component_names(::Type{S}) where S
+
+Project `vals` to the grid nodes with `proj` and save to the stream.
+"""
+function write_projection(vtk::VTKFile, proj::L2Projector, vals, name)
+    data = _evaluate_at_grid_nodes(proj, vals, #=vtk=# Val(true))::Matrix
+    @assert size(data, 2) == getnnodes(get_grid(proj.dh))
+    _vtk_write_nodedata(vtk.vtk, data, name; component_names=component_names(eltype(vals)))
+    return vtk
+end
+
+"""
+    write_celldata(vtk::VTKFile, celldata::AbstractVector, name::String)
+
+Write the `celldata` that is ordered by the cells in the grid to the vtk file.
+"""
+function write_celldata(vtk::VTKFile, celldata, name)
+    WriteVTK.vtk_cell_data(vtk.vtk, celldata, name)
+end
+
+"""
+    write_nodedata(vtk::VTKFile, nodedata::Vector{Real}, name)
+    write_nodedata(vtk::VTKFile, nodedata::Vector{<:AbstractTensor}, name)
+    
+Write the `nodedata` that is ordered by the nodes in the grid to the vtk stream.
+
+When `nodedata` contains `Tensors.Vec`s, each component is exported. 
+Two-dimensional vectors are padded with zeros.
+
+When `nodedata` contains second order tensors, the index order, 
+`[11, 22, 33, 23, 13, 12, 32, 31, 21]`, follows the default Voigt order in Tensors.jl.
+"""
+function write_nodedata(vtk::VTKFile, nodedata, name)
+    _vtk_write_nodedata(vtk.vtk, nodedata, name)
+end
+
+
+"""
+    write_nodeset(vtk::VTKFile, grid::AbstractGrid, nodeset::String)
+
+Write nodal values of 1 for nodes in `nodeset`, and 0 otherwise
+"""
+function write_nodeset(vtk, grid::AbstractGrid, nodeset::String)
     z = zeros(getnnodes(grid))
     z[collect(getnodeset(grid, nodeset))] .= 1.0
-    vtk_point_data(vtk, z, nodeset)
+    write_nodedata(vtk, z, nodeset)
+    return vtk
 end
 
 """
-    vtk_cellset(vtk, grid::Grid)
+    write_cellset(vtk, grid::AbstractGrid)
+    write_cellset(vtk, grid::AbstractGrid, cellset::String)
+    write_cellset(vtk, grid::AbstractGrid, cellsets::Union{AbstractVector{String},AbstractSet{String})
 
-Export all cell sets in the grid. Each cell set is exported with
-`vtk_cell_data` with value 1 if the cell is in the set, and 0 otherwise.
+Write all cell sets in the grid with name according to their keys and 
+celldata 1 if the cell is in the set, and 0 otherwise. It is also possible to 
+only export a single `cellset`, or multiple `cellsets`. 
 """
-function vtk_cellset(vtk::WriteVTK.DatasetFile, grid::AbstractGrid, cellsets=keys(grid.cellsets))
+function write_cellset(vtk, grid::AbstractGrid, cellsets=keys(getcellsets(getgrid(vtk))))
     z = zeros(getncells(grid))
     for cellset in cellsets
-        z .= 0.0
+        fill!(z, 0)
         z[collect(getcellset(grid, cellset))] .= 1.0
-        vtk_cell_data(vtk, z, cellset)
+        write_celldata(vtk, z, cellset)
     end
     return vtk
 end
+write_cellset(vtk, grid::AbstractGrid, cellset::String) = write_cellset(vtk, grid, [cellset])
 
 """
-    vtk_cellset(vtk, grid::Grid, cellset::String)
+    write_dirichlet(vtk::VTKFile, ch::ConstraintHandler)
```

Comment:
Perhaps `write_constraints` and we can add a field where affine-constrained dofs are 1? (Can be in a separate PR, but perhaps `write_constriants` could be changed here anyway)

Reply:
#920